cmd/deployment/create.go

```diff
@@ -0,0 +1,169 @@
+                                "ml": false
+                            },
+                            "zone_count": 1
+                        }
+                    ]
+                }
+            }
+        ],
+        "kibana": [
+            {
+                "display_name": "my kibana instance",
+                "ref_id": "my-kibana-instance",
+                "elasticsearch_cluster_ref_id": "my-es-cluster",
+                "plan": {
+                    "zone_count": 1,
+                    "kibana": {
+                        "version": "6.8.4"
+                    },
+                    "cluster_topology": [
+                        {
+                            "instance_configuration_id": "kibana",
+                            "memory_per_node": 1024,
+                            "node_count_per_zone": 1
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+}
+$ ecctl deployment create -f deployment_example.json --version=7.4.1
+[...]`[1:]
+
+var createCmd = &cobra.Command{
+	Use:     `create -f <file definition.json>`,
+	Short:   "Creates a deployment from a file definition, allowing certain flag overrides",
+	Long:    createLong,
+	Example: createExample,
+	PreRunE: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		filename, _ := cmd.Flags().GetString("file")
+		var r models.DeploymentCreateRequest
+		if err := cmdutil.DecodeFile(filename, &r); err != nil {
+			return err
+		}
+
+		var region string
+		if ecctl.Get().Config.Region == "" {
+			region = cmdutil.DefaultECERegion
+		}
+
+		name, _ := cmd.Flags().GetString("name")
+		version, _ := cmd.Flags().GetString("version")
+
+		reqID, _ := cmd.Flags().GetString("request-id")
+		if reqID == "" {
+			reqID = cmdutil.RandomString(64)
+		}
+
+		res, err := deployment.Create(deployment.CreateParams{
+			API:       ecctl.Get().API,
+			RequestID: reqID,
+			Request:   &r,
+			Overrides: &deployment.CreateOverrides{
+				Name:    name,
+				Region:  region,
+				Version: version,
+			},
+		})
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr,
+				"The deployment creation returned with an error, please use the displayed idempotency token to recreate the deployment resources",
+			)
+			fmt.Fprintln(os.Stderr, "Idempotency token:", reqID)
+			return err
+		}
+
+		return ecctl.Get().Formatter.Format("", res)
+	},
+}
+
+func init() {
+	Command.AddCommand(createCmd)
+	createCmd.Flags().String("name", "", "Overrides the deployment name")
+	createCmd.Flags().String("version", "", "Overrides all thee deployment's resources to the specified version")
+	createCmd.Flags().String("request-id", "", "Optional idempotency token - if two create requests share the same request_id token (min size 32 characters, max 128) then only one deployment will be created, the second request will return the info of that deployment (in the same format described below, but with blanks for auth-related fields)")
```

Comment:
I assume users know how to generate this token? Should we explain somehow? Or point them to some online documentation? For instance I have no idea what is this and how to generate it :)

Reply:
I can change the wording on this flag, but an explanation on the indempotency tokens can be found in line :32-34 and here: https://github.com/elastic/ecctl/pull/36/files#diff-437f06621adf011db767be2048d80d9eR7-R10

Reply:
Let me know if f8842ac addresses your concerns

Reply:
Also th PR description has some info about it

Reply:
Yeah now it's super clear! 🙇 👍